compiler/passes/checkResolved.cpp

```diff
@@ -541,13 +541,15 @@ static void
 checkReturnPaths(FnSymbol* fn) {
   // Check to see if the function returns a value.
   if (fn->isIterator() ||
+      fn->hasFlag(FLAG_THUNK_BUILDER) ||
```

Comment:
Surprisingly, yes -- `isIterator` is the iterator record analog to `FLAG_THUNK_BUILDER` (function that returns an iterator / function that returns a thunk record) whereas `FLAG_AUTO_II` is the iterator analog to `FLAG_THUNK_INVOKE` (methods like `zip` etc / method `invoke`). So I put them near their analogs.

Reply:
I see. Could you put comments next to them pointing those similarities out? This could strike someone as odd, and comments could help understand things a bit better.